applications/nrf_desktop/src/modules/config.c

```diff
@@ -150,6 +157,47 @@ static void update_config(const u8_t config_id, u8_t *data, size_t size)
 	}
 }
 
+static void settings_load_thread(void)
+{
+	LOG_INF("Settings load thread started");
+
+	int err = 0;
+
+	err = settings_load();
+
+	if (err) {
+		module_set_state(MODULE_STATE_ERROR);
+		LOG_ERR("Cannot load settings");
+	} else {
+		module_set_state(MODULE_STATE_READY);
+		LOG_INF("Settings loaded");
+	}
+}
+
+static void start_loading_thread(void)
+{
+	if (atomic_get(&thread_active)) {
+		return;
+	}
+
+	k_thread_create(&thread, thread_stack, THREAD_STACK_SIZE,
+					(k_thread_entry_t)settings_load_thread,
+					NULL, NULL, NULL,
+					THREAD_PRIORITY, 0, K_NO_WAIT);
+	k_thread_name_set(&thread, MODULE_NAME "_thread");
+
+	atomic_set(&thread_active, true);
+}
+
+static void thread_cleanup(void)
+{
+	if (!atomic_get(&thread_active)) {
```

Comment:
to be considered

Reply:
I found only one function that returns state of the thread k_thread_state_str that returns string. I think it is better to use atomic instead of string comparison.